ros2topic/ros2topic/verb/info.py

```diff
@@ -12,19 +12,43 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from rclpy.qos import QoSDurabilityPolicy
+from rclpy.qos import QoSLivelinessPolicy
+from rclpy.qos import QoSReliabilityPolicy
+
 from ros2cli.node.direct import DirectNode
 from ros2topic.api import get_topic_names_and_types
 from ros2topic.api import TopicNameCompleter
 from ros2topic.verb import VerbExtension
 
 
+def print_topic_endpoint_info(topic_name, get_topic_endpoint_info_func):
```

Comment:
I left a comment on the `rclpy` side, but it seems like `node.get_publishers_info_by_topic()`/`node.get_subscriptions_info_by_topic()` should return a list of objects rather than just dict's and then that class could have a to string implementation. That way this kind of printing logic could be reused. Also it would help when adding fields to the info. Right now who ever is adding information has to know that this code exists, otherwise the new fields will not be printed.

Reply:
Done.